debezium-core/src/main/java/io/debezium/relational/history/KafkaDatabaseHistory.java

```diff
@@ -314,8 +364,76 @@ public boolean exists() {
+                    return;
+                }
+                final Config topic = configs.values().iterator().next();
+                if (topic == null) {
+                    LOGGER.info("Could not get config for topic '{}'", topic);
+                    return;
+                }
+
+                final String cleanupPolicy = topic.get(CLEANUP_POLICY_NAME).value();
+                if (!CLEANUP_POLICY_VALUE.equals(cleanupPolicy)) {
+                    LOGGER.warn("Database history topic '{}' option '{}' should be '{}' but is '{}'", topicName, CLEANUP_POLICY_NAME, CLEANUP_POLICY_VALUE,
+                            cleanupPolicy);
+                    return;
+                }
+
+                final String retentionBytes = topic.get(RETENTION_BYTES_NAME).value();
+                if (retentionBytes != null && Long.parseLong(retentionBytes) != UNLIMITED_VALUE) {
+                    LOGGER.warn("Database history topic '{}' option '{}' should be '{}' but is '{}'", topicName, RETENTION_BYTES_NAME, UNLIMITED_VALUE, retentionBytes);
+                    return;
+                }
+
+                final String retentionMs = topic.get(RETENTION_MS_NAME).value();
+                if (retentionMs != null && (Long.parseLong(retentionMs) != UNLIMITED_VALUE && Long.parseLong(retentionMs) < RETENTION_MS_MIN)) {
+                    LOGGER.warn("Database history topic '{}' option '{}' should be '{}' or greater than '{}' (5 years) but is '{}'", topicName, RETENTION_MS_NAME,
+                            UNLIMITED_VALUE, RETENTION_MS_MIN, retentionMs);
+                    return;
+                }
+
+                final DescribeTopicsResult result = admin.describeTopics(Collections.singleton(topicName));
+                if (result.values().size() != 1) {
+                    LOGGER.info("Expected one topic '{}' to match the query but got {}", topicName, result.values().size());
+                    return;
+                }
+                final TopicDescription topicDesc = result.values().values().iterator().next().get();
+                if (topicDesc == null) {
+                    LOGGER.info("Could not get description for topic '{}'", topicName);
+                    return;
+                }
+
+                final int partitions = topicDesc.partitions().size();
+                if (partitions != PARTITION_COUNT) {
+                    LOGGER.warn("Database history topic '{}' should have one partiton but has '{}'", topicName, partitions);
+                    return;
+                }
+
+                LOGGER.info("Database history topic '{}' has correct settings", topicName);
+            }
+            catch (Throwable e) {
+                LOGGER.info("Attempted to validate database history topic but failed", e);
+            }
+        });
+    }
+
     @Override
     public synchronized void stop() {
+        if (checkTopicSettingsExecutor != null) {
```

Comment:
Ok, last comment on this PR, promised :) Can we move the shutdown right to the topic check (with awaitTermination()), so we don't keep it running for all the time.

Reply:
@gunnarmorling Done, but I don't believe you ;-)

Reply:
Let me see... ;-)